src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -343,7 +343,7 @@ public bool RestartListener
     /// having correct access restrictions, and provides a listener
     /// thread loop.
     /// </summary>
-    internal sealed class RemoteSessionNamedPipeServer : IDisposable
+    public sealed class RemoteSessionNamedPipeServer : IDisposable
```

Comment:
Why was it changed to public? It seems it is used only in tests - for tests we could use a test hook.

We need PowerShell Committee approval for new public API.

Reply:
This is needed for hosted PowerShell scenarios.  PS-Committee is not required to approve all public APIs, only if there is uncertainty or controversy.  As long as a PS-Maintainer is ok with the public API, we should not have the PS-Committee be a bottleneck.

Reply:
> This is needed for hosted PowerShell scenarios. 

For hosted scenarios I believe it'll be good to have xUnit tests.

Reply:
I've moved those select tests over to xUnit